### DISCUSSION TITLE
Can I prevent kitty from modifying my PATH when it starts?

### DISCUSSION THREAD

**Discussion by jpnance** [kovidgoyal/kitty#6626] [Q&A]
Context: MacOS, bash, yabai

Reference: https://sw.kovidgoyal.net/kitty/faq/#how-do-i-specify-command-line-options-for-kitty-on-macos

I've spent a good amount of time trying to debug why nvm doesn't seem to want to remember across shell sessions when I tell it what I'd like its "default" version of Node to be and I think it's something to do with how kitty injects itself into my `PATH` variable when I start it up. When I open a new Apple Terminal session, for instance, and run `node -v`, I get the version number I'd like to see, the new default; same thing with a new iTerm session. When I open a new kitty session, though, `node -v` still shows me the old default and I have to explicitly do `nvm use default` to switch to the new one. nvm makes use of the `PATH` variable and, indeed, when I'm in kitty, I see that my `PATH` references the old default Node version that I installed. This behavior has persisted across quitting all kitty sessions and even restarting my computer.

How does it work when kitty is placing itself into my `PATH`? Can I disable that behavior if I think it's causing problems? I've tried searching the codebase but I can't find what I think I'm looking for. For what it's worth, the main way I open a new kitty terminal is via a shortcut (configured with skhd) that does `open -a kitty ~`.

Please let me know if there's any more context that I can provide to help!

**Comment by kovidgoyal** [comment]
There is no way that kitty inserting itself into PATH can cause what you
describe. And if you want to disable it look at the
ensure_kitty_in_path() function.

**Comment by jpnance** [comment]
I've done some more digging and I was able to eventually get iTerm to exhibit the same behavior so you're right that this isn't strictly a kitty issue. I understand if you'd rather close this discussion at this point.

I've noticed that my `PATH` in kitty is different depending on how I initially open up the application. If I restart my computer, then click the kitty icon in the dock, then do `node -v`, I see the Node version number that I want to see. If, however, I initially launch kitty by using my keyboard shortcut (which runs the command `open -a kitty ~`), I end up seeing the wrong Node version number. Once kitty is started, all subsequent terminal windows seem to use whatever `PATH` the first one used. This same exact behavior seems to occur in iTerm but (very strangely) *not* in Terminal! Terminal appears to always get my `PATH` right, no matter how I initially open it up.

At this point, I guess I'm mostly interested in what the difference is between clicking the dock and running `open`. I get that MacOS internals aren't necessarily in your purview, so I just leave this information here in case it helps you (or anyone else) understand what's going on.

Thanks for the reply and for the wonderful terminal emulator!

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/faq/#things-behave-differently-when-running-kitty-from-system-launcher-vs-from-another-terminal
